python/ray/autoscaler/aws/node_provider.py

```diff
@@ -79,40 +133,25 @@ def is_terminated(self, node_id):
         return state not in ["running", "pending"]
 
     def node_tags(self, node_id):
-        node = self._node(node_id)
-        tags = {}
-        for tag in node.tags:
-            tags[tag["Key"]] = tag["Value"]
-        return from_aws_format(tags)
+        with self.tag_cache_lock:
+            d1 = self.tag_cache[node_id]
+            d2 = self.tag_cache_pending.get(node_id, {})
+            return {**d1, **d2}
```

Comment:
```suggestion
            return dict(d1, **d2)
```

Seems like this doesn't work in python 2